source/menus/utils.py

```diff
@@ -29,14 +20,49 @@ def render(self, term: Interface) -> str:
         ...
 
     def kinput(self, term: Interface, key: Union[int, None]) -> None:
-        """Handle any keyboard input, this method should return a new selected integer."""
+        """Handle any keyboard input however the menu see fit."""
         ...
 
-    def click(self, term: Interface) -> State:
-        """Handle ENTER presses, this method should return a new State enum."""
+    def click(self, term: Interface) -> 'Menu':  # Forward reference
+        """Handle ENTER presses, this method should return a menu instance."""
         ...
 
 
+class PopupMessage(Menu):
+    """Toast message class"""
+
+    def __init__(self, origin: Menu, message: str, color: Callable) -> None:
+        """Initialize the toast message"""
+        self.origin = origin
+        self.message = message
+        self.color = color
+
+    def render(self, term: Interface) -> str:
+        """Render the message waiting for the user to continue"""
+        width = min(term.width - 8, len(self.message) + 12) - 2
+        toast = term.center('┌' + '—' * width + '┐') + '\n'
+        toast += term.center('│' + ' ' * width + '│') + '\n'
+        toast += term.center(
+            '│' + math.floor((width - len(self.message)) // 2) * ' '
+            + self.color(self.message)
+            + math.ceil((width - len(self.message)) // 2) * ' ' + '│'
+        ) + '\n'
+        toast += term.center('│' + ' ' * width + '│') + '\n'
+        toast += term.center(
+            '│' + math.floor((width - 23) // 2) * ' '
+            + 'Press enter to continue'
+            + math.ceil((width - 23) // 2) * ' ' + '│'
+        ) + '\n'
+        toast += term.center('│' + ' ' * width + '│') + '\n'
+        toast += term.center('└' + '-' * width + '┘')
+
+        return term.move_y(term.height // 2) + toast
```

Comment:
Instead of having to add `\n` to the end of each line. What about keeping it as a list of strings and then doing this at the end:
```suggestion
        return term.move_y(term.height // 2) + '\n'.join(toast)
```